File: cpp/box.cpp
#include <iostream>
#include <utility>

template<typename T>
class Box {
    T value;
public:
    Box(T value) : value(std::move(value)) {}

    T getValue() {
        return value;
    }
};

int main() {
    auto wrapped = new Box{50};
    auto wrapped_string = new Box{std::string{"abcd"}};

    std::cout << wrapped->getValue() << std::endl;
    std::cout << wrapped_string->getValue() << std::endl;

    delete wrapped;
    delete wrapped_string;
    return 0;
}

File: java/src/compiled/simple/Box.java
package compiled.simple;

public class Box {

    public Box(Object value) {
        this.value = value;
    }

    public Object getValue() {
        return value;
    }

    public static void main(String args[]) {
        Box wrapped = new Box(Integer.valueOf(10));
        Box wrappedString = new Box("abcd");
        System.out.println(wrapped.getValue());
        System.out.println((String) wrappedString.getValue());
    }

    private Object value;
}
